torch/ao/quantization/backend_config/executorch.py

```diff
@@ -0,0 +1,216 @@
+        # Tensor dtype inference properly
+        0: ObservationType.OUTPUT_USE_DIFFERENT_OBSERVER_AS_INPUT,
+        1: ObservationType.OUTPUT_SHARE_OBSERVER_WITH_INPUT,
+        2: ObservationType.OUTPUT_USE_DIFFERENT_OBSERVER_AS_INPUT,
+    }
+    binary_op_configs: List[BackendPatternConfig] = []
+    for op in [operator.add, torch.add]:
+        binary_op_configs.append(
+            BackendPatternConfig(op)
+                .set_dtype_configs(dtype_configs)  # noqa: E131
+                ._set_num_tensor_args_to_observation_type(num_tensor_args_to_observation_type_mapping))
+    return binary_op_configs
+
+def _get_share_qparams_ops_configs() -> List[BackendPatternConfig]:
+    """
+    Return the operator configs for the operators that works for both float and quantized
+    input if input is quantized, the output Tensor shares the same quantization parameter
+    with input.
+
+    Example operator: avgpool2d, reshape, transpose, maxpool2d
+    Example observed operator:
+    observer_0 - avgpool2d - observer_0 (same observer instance as input)
+    """
+    observation_type = ObservationType.OUTPUT_SHARE_OBSERVER_WITH_INPUT
+    dtype_configs = [executorch_default_op_quint8_dtype_config]
+    share_qparams_ops = [
+        F.adaptive_avg_pool2d,
+        F.relu,
+        F.relu6,
+        torch.nn.AdaptiveAvgPool2d,
+        torch.squeeze,
+        "permute",
+        "reshape",
+        "relu",
+        "relu_",
+        "squeeze",
+        "squeeze_",
+    ]
+    share_qparams_op_configs: List[BackendPatternConfig] = []
+    for op in share_qparams_ops:
+        share_qparams_ops.append(
+            BackendPatternConfig(op)
+                .set_observation_type(observation_type)  # noqa: E131
+                .set_dtype_configs(dtype_configs))
+    return share_qparams_op_configs
+
+def _get_bn_configs() -> List[BackendPatternConfig]:
+    """
+    Return all configs related to batchnorm.
+    """
+    observation_type = ObservationType.OUTPUT_USE_DIFFERENT_OBSERVER_AS_INPUT
+    dtype_configs = [executorch_default_op_quint8_dtype_config]
+    bn_configs = []
+    bn_configs.append(
+        BackendPatternConfig(nn.BatchNorm2d)
+            .set_observation_type(observation_type)  # noqa: E131
+            .set_dtype_configs(dtype_configs))
+    return bn_configs
+
+
+# =====================
+# |  BACKEND CONFIGS  |
+# =====================
+
+def get_executorch_backend_config() -> BackendConfig:
+    """
+    Return the `BackendConfig` for backends PyTorch lowers to through the Executorch stack.
+    """
+    return BackendConfig("executorch") \
+        .set_backend_pattern_configs(_get_linear_configs()) \
+        .set_backend_pattern_configs(_get_conv_configs()) \
+        .set_backend_pattern_configs(_get_binary_ops_configs()) \
+        .set_backend_pattern_configs(_get_share_qparams_ops_configs()) \
+        .set_backend_pattern_configs(_get_bn_configs())
+
+__all__ = [
```

Comment:
should this appear in the top of the file?

Reply:
Oh, yes. I just copied it over from FBGEMM. I'll move it